craft_application/application.py

```diff
@@ -236,7 +236,19 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
         """
         if project_dir is None:
             project_dir = self.project_dir
-        return (project_dir / f"{self.app.name}.yaml").resolve(strict=True)
+
```

Comment:
In light of the way Snapcraft overrides this method I'm thinking that it might be preferable to let this `_resolve_project_path()` method keep raising a `FileNotFoundError`, and then catch it in `get_project()` to re-raise as `ProjectFileMissingError`. The reason is that Snapcraft is already [relying on this raising a FileNotFound](https://github.com/canonical/snapcraft/blob/feature/craft-application/snapcraft/application.py#L81) so that code would need to be updated too
(also we should test this branch on Snapcraft)

Reply:
I plan to update the snapcraft code for this, so it should be fine. And this also allow others to get a better error even if the app not override it.

Reply:
ok, then I guess I'd like to see what you mean on the Snapcraft side

Reply:
Let's move this to `get_project` so most apps don't need to catch the error even if they do override this method. Combined with @tigarmo's other suggestion we probably don't have to do anything to snapcraft for this at the moment. If users get confused we can provide some snapcraft-specific logic.